src/esp/bindings_js/bindings_js.cpp

```diff
@@ -92,10 +96,45 @@ EMSCRIPTEN_BINDINGS(habitat_sim_bindings_js) {
       .element(em::index<2>())
       .element(em::index<3>());
 
+  em::value_array<Magnum::Vector2i>("Vector2i")
+      .element(em::index<0>())
+      .element(em::index<1>());
+
+  em::class_<Magnum::Matrix4>("Matrix4")
```

Comment:
@mosra Is this really the best way to bind Magnum's Matrices and Vectors? Surely there is a better way.

Reply:
What do you mean, exactly? Can you elaborate? To me this looks like bindings done on an as-needed basis, and for that it looks fine.

For a general case of "expose everything possible" this could get templated by size and type, similarly to how Python bindings are done. But JavaScript has no operator overloads nor anything similar so any custom math code will always be extremely miserable.

Reply:
I guess I was thinking of the expose everything possible template, but if you say we don't need it yet, I suppose we don't have to go down the root just yet.